server/jetstream_test.go

```diff
@@ -4576,6 +4576,41 @@ func TestJetStreamSnapshotsAPI(t *testing.T) {
 	}
 }
 
+func TestJetStreamPubAckFutureComplete(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	if config := s.JetStreamConfig(); config != nil {
+		defer removeDir(t, config.StoreDir)
+	}
+	defer s.Shutdown()
+
+	nc, js := jsClientConnect(t, s)
+	defer nc.Close()
+
+	if _, err := js.AddStream(&nats.StreamConfig{Name: "foo", Storage: nats.MemoryStorage}); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	toSend := 1_000_000
```

Comment:
Sending that many messages when running under `-race` is not good. So you should move it to the `norace_test.go`, but more importantly, is this a test to show that there is a server issue or a client issue? If client (since I saw that you opened an issue in the nats.go repo), then I would not add it this in server test.

Reply:
I put this here because I clone-updated an existing test in this file.
This test fails now, so don't want to merge this. We can close it. It was just a way to share test code with you.

I don't know if this is a client or server issue. I'm just reporting a problem I noticed.